tests/unit_tests/test_kmers.py

```diff
@@ -158,3 +185,64 @@ def test_embed_out_exists(norm_df, tmp_path):
         method=method,
         seed=seed,
     )
+
+
+def test_embed_TableFormatError(invalid_df_fpath):
+    with pytest.raises(TableFormatError):
+        kmers.embed(kmers=invalid_df_fpath)
+
+
+def test_embed_TypeError(tmp_path):
+    kmer_fpath = tmp_path / "kmers.embed.tsv"
+    with pytest.raises(TypeError):
+        kmers.embed(kmers=kmer_fpath)
+
+
+@patch("os.path.getsize", return_value=2 * 1024 * 1024)
```

Comment:
I think we should not use multiple different patching and test methods/frameworks. This adds complexity to the testing suite and forces other developers to learn more frameworks than is necessary. It is not clear what this patch is doing or why it is necessary.

```suggestion

```